### PR TITLE
add some initial lightweight testing

### PR DESCRIPTION
**Pull request by lindsaylevine**
this also addresses https://github.com/netlify/netlify-plugin-nextjs/issues/8 and https://github.com/netlify/netlify-plugin-nextjs/issues/12

1. we see that exact error in issue #8 running tests if next isnt installed in the project. should i try to mock it? :/
2. seems we don't need hasFramework anymore with the next require try/catch. also when this is upgraded to a core plugin, won't the framework check be done outside of the plugin?
3. tried to mock `fs` in the standard way i mock the other dependencies and ran into a handful of issues. i now use the mock-fs package but calling mockFs broke the internals of `hasFramework` despite efforts to pass a correct config to mockFs that would appease hasFramework ☹️

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,7 +53,7 @@ module.exports = {
             target: 'serverless'
           }
         `
-      await appendFile('next.config.js', nextConfig)
+      appendFileSync('next.config.js', nextConfig)
```
Comment (ehmicky) [inline index.js:56]: Note: if you want to keep it async while maintaining old Node versions (i.e. without `fs.promise`), you could also use:

```js
const { promisify } = require('util')
const appendFile = promisify(fs.appendFile)
```

File: index.js [hunk 2]
```diff
@@ -15,13 +12,8 @@ const isStaticExportProject = require('./helpers/isStaticExportProject')
 // - Between the build and postbuild steps, any functions are bundled
 
 module.exports = {
-  async onPreBuild({ netlifyConfig, packageJson: { scripts = {}, dependencies = {} }, utils }) {
+  async onPreBuild({ netlifyConfig = {}, packageJson: { scripts = {}, dependencies = {} }, utils }) {
     const { failBuild } = utils.build
-
-    if (!(await hasFramework('next'))) {
```
Comment (ehmicky) [inline index.js:21]: Until this is made a core plugin, if a user is installing this plugin but they forgot to install `next`, wouldn't it be better to print this error message?

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "format": "prettier --write .",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "jest"
```
Comment (ehmicky) [inline package.json:8]: Should we also add some GitHub actions, so tests are run when someone creates a PR?

File: package.json [hunk 4]
```diff
@@ -23,19 +23,27 @@
   },
   "homepage": "https://github.com/netlify/netlify-plugin-nextjs#readme",
   "dependencies": {
-    "@netlify/framework-info": "^0.2.0",
     "cpx": "^1.5.0",
     "make-dir": "^3.1.0",
+    "next": "^10.0.1",
```
Comment (ehmicky) [inline package.json:28]: What if the site is using a different version of next, e.g. `9.0.0`?
In that case, this plugin would assume the site is using a different version.

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
+const path = require('path')
```
Comment (ehmicky) [inline test/index.js:1]: Nice! :clap:

File: package.json (hunk 4 above)
Comment (lindsaylevine) [reply package.json:28]: oof true

Comment (lindsaylevine) [reply package.json:28]: this version basically just needs to be aligned with next-on-netlify, which hardcodes its version as  `"next": "9.x"` 

in the future NoN will likely just live in here and so we won't have to manage in multiple places

**Comment by lindsaylevine** [comment]
now fixes https://github.com/netlify/netlify-plugin-nextjs/issues/12

**Pull request merged**